internal/model/matcher.go

```diff
@@ -43,6 +45,51 @@ func NewSimpleFileMatcher(paths ...string) (fileMatcher, error) {
 	return fileMatcher{paths: pathMap}, nil
 }
 
+// NewRelativeFileMatcher returns a matcher for the given paths; any relative paths
+// are converted to absolute, relative to the given baseDir.
+func NewRelativeFileMatcher(baseDir string, paths ...string) fileMatcher {
+	pathMap := make(map[string]bool, len(paths))
+	for _, path := range paths {
+		if !filepath.IsAbs(path) {
+			path = filepath.Join(baseDir, path)
+		}
+		pathMap[path] = true
+	}
+	return fileMatcher{paths: pathMap}
+}
+
+// A PathSet stores one or more filepaths, along with the directory that any
+// relative paths are relative to
+type PathSet struct {
+	Paths         []string
+	BaseDirectory string
+}
+
+func NewPathSet(paths []string, baseDir string) PathSet {
+	return PathSet{
+		Paths:         paths,
+		BaseDirectory: baseDir,
+	}
+}
+
+func (ps PathSet) Empty() bool { return len(ps.Paths) == 0 }
+
+// AnyMatch returns true if any of the given filepaths match any paths contained in the pathset.
+func (ps PathSet) AnyMatch(paths []string) (bool, error) {
+	matcher := NewRelativeFileMatcher(ps.BaseDirectory, ps.Paths...)
```

Comment:
i thought the plan was to use the DockerPatternMatcher to preserve the existing behavior? this changes the behavior...right?

Reply:
Iiiiit _does_, but it still supports all of the behavior we ADVERTISE ourselves as supporting. the fact that we use dockerIgnore under the hood is really weird, considering that we only ever talk about exact path matches.